.github/workflows/release-milestone.yaml

```diff
@@ -0,0 +1,66 @@
+name: 'Release Milestone Check'
+on:
+  pull_request_target:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - labeled
+    - unlabeled
+
+jobs:
+  check-release-milestone:
+    name: Check Release Milestone
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        path: './'
+
+    - id: version
+      run: |
+        echo "::set-output name=version::$(cat ./VERSION)"
+
+    - name: Block merge if release milestone is missing
+      uses: actions/github-script@v4
+      with:
+        github-token: ${{ secrets.GITHUB_TOKEN }}
+        script: |
+          const nextMinorRegex = /^(v[0-9]+\.[0-9]+)\.0-dev/gmi;
+
+          let match = nextMinorRegex.exec('${{ steps.version.outputs.version }}');
+          if (match == null) {
+            core.info('VERSION does not indicate that the next version is a new minor release - skipping check');
+            return;
+          }
+          const nextMinorVersion = match[1];
+
+          const milestones = await github.issues.listMilestones({
+            owner: context.repo.owner,
+            repo: context.repo.repo,
+            state: 'open',
+            sort: 'due_on',
+            direction: 'desc'
+          });
+
+          let milestoneForNextMinorReleaseFound = false;
+
+          for (const milestone of milestones.data) {
+            if (milestone.title != nextMinorVersion) {
+              continue;
+            }
+
+            // milestone found, check if PR is associated
+            milestoneForNextMinorReleaseFound = true;
+
+            if (context.payload.pull_request.milestone == null) {
```

Comment:
What about the case, that the PR is marked for a different milestone?

Reply:
I also thought about it, but I think this case doesn't exist today in our workflow, so I went for ignoring it to keep the script simple. WDYT?

Reply:
Fair enough :)